client/src/views/files/FileOperationMenu.vue

```diff
@@ -232,24 +197,34 @@ function getFileOperationComponent(item: OperationItem): Component | undefined {
 }
 
 function updateImportState(id: string, state: FileOperationState, stateData?: StateData): void {
-  const operation = fileOperations.value.find((op) => op.data.id === id);
+  const operation = inProgressItems.value.find((op) => op.data.id === id);
```

Comment:
You could use `findIndex` right here and use it to get the matching operation, just to avoid going through the list againt when removing the item.